README.md

```diff
@@ -1,72 +1,185 @@
-maintained packages.
+azure-sdk-for-go provides Go packages for using Azure services.
 
-> **NOTE:** This repository is under heavy ongoing development and should be considered a preview. Vendoring your
-dependencies is always a good idea, but it is doubly important if you're consuming this library.
+:exclamation: **NOTE:** This project is in preview and breaking changes are 
+introduced frequently. Therefore, vendoring dependencies is even more
+important than usual. We use [dep](https://github.com/golang/dep).
 
-# Installation
-- If you don't already have it, install [the Go Programming Language](https://golang.org/dl/).
-- Go get the SDK:
+To be notified about updates and changes, subscribe to the [Azure update feed][].
 
-```
+**Install:**
+
+```sh
 $ go get -u github.com/Azure/azure-sdk-for-go/...
 ```
 
-> **IMPORTANT:** We highly suggest vendoring Azure SDK for Go as a dependency. For vendoring dependencies, Azure SDK
-for Go uses [dep](https://github.com/golang/dep).
+If you need to install Go, follow [the official instructions][].
+
+[the official instructions]: https://golang.org/dl/
+
+**Use:**
+
+For complete examples see [Azure-Samples/azure-sdk-for-go-samples][samples_repo].
+
+1. Import a package from the [services][services_dir] directory. See [SERVICES.md][] for
+   a list.
+1. Create and authenticate a client with a `New*Client` func, e.g.
+   `c := compute.NewVirtualMachinesClient(...)`.
+1. Invoke API methods using the client, e.g. `c.CreateOrUpdate(...)`.
+1. Handle responses.
+
+For example, to create a new virtual network (substitute your own values for
+strings in angle brackets):
+
+```go
+import (
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+vnetClient := network.NewVirtualNetworksClient("<subscriptionID>")
+vnetClient.Authorizer = autorest.NewBearerAuthorizer("<OAuthTokenProvider>")
+
+vnetClient.CreateOrUpdate(
+  "<resourceGroupName>",
+  "<vnetName>",
+  network.VirtualNetwork{
+    Location: to.StringPtr("<azureRegion>"),
+    VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
+      AddressSpace: &network.AddressSpace{
+        AddressPrefixes: &[]string{"10.0.0.0/8"},
+      },
+      Subnets: &[]network.Subnet{
+        {
+          Name: to.StringPtr("<subnet1Name>"),
+          SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+            AddressPrefix: to.StringPtr("10.0.0.0/16"),
+          },
+        },
+        {
+          Name: to.StringPtr("<subnet2Name>"),
+          SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+            AddressPrefix: to.StringPtr("10.1.0.0/16"),
+          },
+        },
+      },
+    },
+  },
+  nil)
+```
+
+## Authentication
+
+Most operations require an OAuth token for authentication and authorization.
+You can get one from Azure AD using the
```

Comment:
Coming this sprint hopefully a less verbose way of auth

Reply:
@sptramer it will still rely on the present authentication methods present in `go-autorest`, they will be a convenience layer to simplify the experience.
A PR with some of the work is here https://github.com/Azure/go-autorest/pull/221 will update it this week after we get an agreement on the best way to expose things, but mostly using what is defined there we would either pass a config to the constructor functions or have a method on them to pass this.